qiita_db/metadata_template/prep_template.py

```diff
@@ -91,6 +91,13 @@ class PrepTemplate(MetadataTemplate):
     _id_column = "prep_template_id"
     _sample_cls = PrepSample
     _filepath_table = 'prep_template_filepath'
+    # This has been explicitly redefined here for clarity, even though
```

Comment:
I'll argue that this is the cool thing about objects, you define it in parent and it's inherit in the children so no need for redefinition. Anyway, I think the actual implementation should be that:
- MetadataTemplate: _forbidden_words = {} or None, {} will be easier
- Leave _forbidden_words in SampleTemplate and PrepTemplate as they are

Reply:
Agreed, however there are tests in test_base_metadata_template.py to test the forbidden words method that wouldn't mean much with set = {}. I can move them to one or both of the sub-class tests, if we're okay with testing parent functionality in children classes. 

Just to round out the discussion, do you think in general, we'll be redefining the set of forbidden words every time we create a new Template? Or do you think in general, new Templates will need to inherit a set of basic forbidden words common to every Template? I ask because they're currently defined as sets, and might be better off as lists, so that they can be appended to.

Reply:
I think each template will have their own rules but there is a chance that some of them will overlap ... now, something interesting is that the main method `_clean_validate_template` has cls._check_subclass() and these auxiliary methods do not. I guess we should add ...

Reply:
Added to _identify_forbidden_words_in_column_names() but not to _identify_pgsql_reserved_words_in_column_names() and _identify_column_names_with_invalid_characters(), since they don't require anything child_specific, and I won't have to move the tests out to one of the child classes.

Reply:
👍